### PR TITLE
fix: update env issues within examples, cosmetic changes to the README, minor fixes within models

### PR DESCRIPTION
**Pull request by aaishikasb**
This PR:
- fixes issues with accessing environment variables in all the sample applications inside examples
- adds cosmetic changes to the README
- updates LLMs for some of the applications

### CODE REVIEW

File: examples/sample_anthropic_stream.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
-client = Anthropic()
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+ANTHROPIC_API_KEY = os.getenv("ANTHROPIC_API_KEY")
 
-th = Toolhouse(provider=Provider.ANTHROPIC)
 
+client = Anthropic(api_key=ANTHROPIC_API_KEY)
+
+# Initialize Toolhouse with the Anthropic provider
+th = Toolhouse(access_token=TH_API_KEY, provider=Provider.ANTHROPIC)
```
Comment (pulphix) [inline examples/sample_anthropic_stream.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_anthropic.py [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
 
-client = Anthropic()
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+ANTHROPIC_API_KEY = os.getenv("ANTHROPIC_API_KEY")
 
-th = Toolhouse(provider=Provider.ANTHROPIC)
+client = Anthropic(api_key=ANTHROPIC_API_KEY)
 
+# Initialize Toolhouse with the Anthropic provider
+th = Toolhouse(access_token=TH_API_KEY, provider=Provider.ANTHROPIC)
```
Comment (pulphix) [inline examples/sample_anthropic.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_groq.py [hunk 3]
```diff
@@ -3,35 +3,39 @@
...
-TOKEN = os.getenv("GROQCLOUD_API_KEY")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+GROQ_TOKEN = os.getenv("GROQCLOUD_API_KEY")
 
-client = Groq(api_key=TOKEN)
+client = Groq(api_key=GROQ_TOKEN)
 
-th = Toolhouse(provider="openai")
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_groq.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_groq_openai_client.py [hunk 4]
```diff
@@ -3,38 +3,42 @@
...
 
 client = OpenAI(
-    api_key=TOKEN,
+    api_key=GROQ_TOKEN,
     base_url="https://api.groq.com/openai/v1"
 )
 
-th = Toolhouse(provider="openai")
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_groq_openai_client.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_groq_openai_client_stream.py [hunk 5]
```diff
@@ -3,31 +3,33 @@
...
 
 client = OpenAI(
-    api_key=TOKEN,
+    api_key=GROQ_TOKEN,
     base_url="https://api.groq.com/openai/v1",
 )
 
-th = Toolhouse(provider="openai")
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_groq_openai_client_stream.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_groq_stream.py [hunk 6]
```diff
@@ -3,28 +3,30 @@
...
-TOKEN = os.getenv("GROQCLOUD_API_KEY")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+GROQ_TOKEN = os.getenv("GROQCLOUD_API_KEY")
 
-client = Groq(api_key=TOKEN)
+client = Groq(api_key=GROQ_TOKEN)
 
-th = Toolhouse(provider="openai")
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_groq_stream.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_llamaindex.py [hunk 7]
```diff
@@ -1,21 +1,27 @@
...
+from toolhouse import Toolhouse # Import the Toolhouse SDK
 import dotenv
 
 dotenv.load_dotenv()
 
-th = Toolhouse(provider="llamaindex")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+
+# Initialize Toolhouse with the Llama Index
+th = Toolhouse(access_token=TH_API_KEY, provider="llamaindex")
```
Comment (pulphix) [inline examples/sample_llamaindex.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_openai.py [hunk 8]
```diff
@@ -1,15 +1,21 @@
...
 
-client = OpenAI()
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+OAI_KEY = os.getenv("OPENAI_API_KEY")
+
+client = OpenAI(access_token=OAI_KEY)
 
-th = Toolhouse(provider="openai")
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_openai.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_openai_stream.py [hunk 9]
```diff
@@ -1,28 +1,33 @@
...
 
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+OAI_KEY = os.getenv("OPENAI_API_KEY")
 
-client = OpenAI()
+client = OpenAI(access_token=OAI_KEY)
 
-th = Toolhouse(provider="openai")
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_openai_stream.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_run_local_tools.py [hunk 10]
```diff
@@ -23,9 +26,10 @@
     }
 ]
 
-client = Anthropic()
+client = Anthropic(api_key=ANTHROPIC_API_KEY)
 
-th = Toolhouse(provider=Provider.ANTHROPIC)
+# Initialize Toolhouse with the Anthropic provider
+th = Toolhouse(access_token=TH_API_KEY, provider=Provider.ANTHROPIC)
```
Comment (pulphix) [inline examples/sample_run_local_tools.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_run_local_tools_groq.py [hunk 11]
```diff
@@ -25,7 +26,8 @@
              'required': ['city']
          }}]
 
-th = Toolhouse(provider="openai")
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_run_local_tools_groq.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_run_local_tools_groq_stream.py [hunk 12]
```diff
@@ -29,7 +30,8 @@
     }
 ]
 
-th = Toolhouse(provider="openai")
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_run_local_tools_groq_stream.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_run_local_tools_openai.py [hunk 13]
```diff
@@ -3,11 +3,19 @@
...
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+OAI_KEY = os.getenv("OPENAI_API_KEY")
+
+client = OpenAI(access_token=OAI_KEY)
+
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_run_local_tools_openai.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_run_local_tools_openai_stream.py [hunk 14]
```diff
@@ -1,15 +1,22 @@
...
 load_dotenv()
 
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+OAI_KEY = os.getenv("OPENAI_API_KEY")
+
+client = OpenAI(access_token=OAI_KEY)
 
-client = OpenAI()
+# Initialize Toolhouse with the OpenAI provider
+th = Toolhouse(access_token=TH_API_KEY, provider="openai")
```
Comment (pulphix) [inline examples/sample_run_local_tools_openai_stream.py]: We should use the api_key variable, the access_token has been deprecated

File: examples/sample_run_local_tools_stream.py [hunk 15]
```diff
@@ -22,9 +25,10 @@
     }
 ]
 
-client = Anthropic()
+client = Anthropic(api_key=ANTHROPIC_API_KEY)
 
-th = Toolhouse(provider=Provider.ANTHROPIC)
+# Initialize Toolhouse with the Anthropic provider
+th = Toolhouse(access_token=TH_API_KEY, provider=Provider.ANTHROPIC)
```
Comment (pulphix) [inline examples/sample_run_local_tools_stream.py]: We should use the api_key variable, the access_token has been deprecated

**Comment by aaishikasb** [comment]
The latest commit also addresses the typo identified in #12.

**Comment by VinciGit00** [comment]
Hi @aaishikasb can you add me as co author?

**Comment by aaishikasb** [comment]
> Hi @aaishikasb can you add me as co author?

Can you send me your GitHub email via Discord? Preferably the private one that GitHub provides!

File: README.md [hunk 16]
```diff
@@ -213,11 +247,11 @@ from toolhouse import Toolhouse
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
-th = Toolhouse(provider="openai")
+th = Toolhouse(api_key=TH_API_KEY, provider="openai")
 th.set_metadata("id", "fabio")  # metadata is optional based on the tools you are using
 th.set_metadata("timezone", 5)  # metadata is optional based on the tools you are using
 
-client = OpenAI()
+client = OpenAI(access_token=OAI_KEY)
```
Comment (iamdaniele) [inline README.md]: Change this to

```
client = OpenAI(api_key=OAI_KEY)
```

File: examples/sample_anthropic_stream.py [hunk 17]
```diff
@@ -20,21 +27,24 @@
     }]
 
 with client.messages.stream(
-    model="claude-3-5-sonnet-20240620",
+    model="gpt-4-turbo",
```
Comment (iamdaniele) [inline examples/sample_anthropic_stream.py]: The line you removed was actually the right one (`gpt-4-turbo` is an OpenAI model but this is an Anthropic example). Change this to:

```
model="claude-3-5-sonnet-20240620"
```

File: examples/sample_anthropic_stream.py [hunk 18]
```diff
@@ -20,21 +27,24 @@
...
         if isinstance(block, MessageStopEvent):
+            # Run the tools using the Toolhouse client with the created message
             messages += th.run_tools(block.message)
         elif isinstance(block, TextEvent):
             print(block.text, end="", flush=True)
 
 
 with client.messages.stream(
-            model="claude-3-5-sonnet-20240620",
+            model="gpt-4-turbo",
```
Comment (iamdaniele) [inline examples/sample_anthropic_stream.py]: The line you removed was actually the right one (`gpt-4-turbo` is an OpenAI model but this is an Anthropic example). Change this to:

```
model="claude-3-5-sonnet-20240620"
```

File: examples/sample_groq.py [hunk 19]
```diff
@@ -3,35 +3,39 @@
...
 
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
-TOKEN = os.getenv("GROQCLOUD_API_KEY")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+GROQ_TOKEN = os.getenv("GROQCLOUD_API_KEY")
 
-client = Groq(api_key=TOKEN)
+client = Groq(api_key=GROQ_TOKEN)
```
Comment (iamdaniele) [inline examples/sample_groq.py]: These are API keys, not tokens. API keys are long lived or never expired until invalidated by the users. Tokens are short lived and usually expire automatically.

```
GROQ_API_KEY = os.getenv("GROQCLOUD_API_KEY")
client = Groq(api_key=GROQ_API_KEY)
```

File: examples/sample_groq_openai_client.py [hunk 20]
```diff
@@ -3,38 +3,42 @@
...
 
-TOKEN = os.getenv("GROQCLOUD_API_KEY")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+GROQ_TOKEN = os.getenv("GROQCLOUD_API_KEY")
 
 client = OpenAI(
-    api_key=TOKEN,
+    api_key=GROQ_TOKEN,
     base_url="https://api.groq.com/openai/v1"
 )
```
Comment (iamdaniele) [inline examples/sample_groq_openai_client.py]: These are API keys, not tokens. API keys are long lived or never expired until invalidated by the users. Tokens are short lived and usually expire automatically.

```
GROQ_API_KEY = os.getenv("GROQCLOUD_API_KEY")
client = OpenAI(
   api_key=GROQ_API_KEY
   # Rest of your code
```

File: examples/sample_groq_openai_client_stream.py [hunk 21]
```diff
@@ -3,31 +3,33 @@
...
 
-TOKEN = os.getenv("GROQCLOUD_API_KEY")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+GROQ_TOKEN = os.getenv("GROQCLOUD_API_KEY")
 
 client = OpenAI(
-    api_key=TOKEN,
+    api_key=GROQ_TOKEN,
     base_url="https://api.groq.com/openai/v1",
 )
```
Comment (iamdaniele) [inline examples/sample_groq_openai_client_stream.py]: These are API keys, not tokens. API keys are long lived or never expired until invalidated by the users. Tokens are short lived and usually expire automatically.

```
GROQ_API_KEY = os.getenv("GROQCLOUD_API_KEY")
client = OpenAI(
   api_key=GROQ_API_KEY
   # Rest of your code
```

File: examples/sample_groq_stream.py [hunk 22]
```diff
@@ -3,28 +3,30 @@
...
 
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
-TOKEN = os.getenv("GROQCLOUD_API_KEY")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+GROQ_TOKEN = os.getenv("GROQCLOUD_API_KEY")
 
-client = Groq(api_key=TOKEN)
+client = Groq(api_key=GROQ_TOKEN)
```
Comment (iamdaniele) [inline examples/sample_groq_stream.py]: Same as other comment, change `GROQ_TOKEN` to `GROQ_API_KEY` (or similar name, but make sure to use the API_KEY suffix)

File: examples/sample_openai.py [hunk 23]
```diff
@@ -1,15 +1,21 @@
...
+from toolhouse import Toolhouse # Import the Toolhouse SDK
 
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
-client = OpenAI()
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+OAI_KEY = os.getenv("OPENAI_API_KEY")
+
+client = OpenAI(access_token=OAI_KEY)
```
Comment (iamdaniele) [inline examples/sample_openai.py]: `client = OpenAI(api_key=OAI_KEY)`

File: examples/sample_openai_stream.py [hunk 24]
```diff
@@ -1,28 +1,33 @@
...
+from toolhouse.models.Stream import ToolhouseStreamStorage # Import the Toolhouse Stream Storage
 
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+OAI_KEY = os.getenv("OPENAI_API_KEY")
 
-client = OpenAI()
+client = OpenAI(access_token=OAI_KEY)
```
Comment (iamdaniele) [inline examples/sample_openai_stream.py]: `client = OpenAI(api_key=OAI_KEY)`

File: examples/sample_run_local_tools_groq.py [hunk 25]
```diff
@@ -8,7 +8,8 @@
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
-TOKEN = os.getenv("GROQCLOUD_API_KEY")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+GROQ_TOKEN = os.getenv("GROQCLOUD_API_KEY")
```
Comment (iamdaniele) [inline examples/sample_run_local_tools_groq.py]:
```
GROQ_API_KEY = os.getenv("GROQCLOUD_API_KEY")
```

File: examples/sample_run_local_tools_groq.py [hunk 26]
```diff
@@ -36,7 +38,7 @@ def hello_tool(city: str):
     return f"Hello from {city}!!!"
 
 
-client = Groq(api_key=TOKEN)
+client = Groq(api_key=GROQ_TOKEN)
```
Comment (iamdaniele) [inline examples/sample_run_local_tools_groq.py]:
```
client = Groq(api_key=GROQ_API_KEY)
```

File: examples/sample_run_local_tools_groq_stream.py [hunk 27]
```diff
@@ -4,15 +4,16 @@
...
 
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
-TOKEN = os.getenv("GROQCLOUD_API_KEY")
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+GROQ_TOKEN = os.getenv("GROQCLOUD_API_KEY")
 
-client = Groq(api_key=TOKEN)
+client = Groq(api_key=GROQ_TOKEN)
```
Comment (iamdaniele) [inline examples/sample_run_local_tools_groq_stream.py]: Same as above, replace `TOKEN` for `API_KEY`

File: examples/sample_run_local_tools_groq_stream.py [hunk 28]
```diff
@@ -44,7 +46,7 @@ def hello_tool(city: str):
 
 
 stream = client.chat.completions.create(
-    model="gpt-4o", messages=messages, tools=th.get_tools() + local_tools, stream=True
+    model="gpt-4o", messages=messages, tools=th.get_tools() + local_tools, stream=True # Retrieve tools installed from Toolhouse
```
Comment (iamdaniele) [inline examples/sample_run_local_tools_groq_stream.py]: `gpt-4o` is an OpenAI model but the code uses Groq, and Groq does not serve OpenAI models. You should change this to any of the [Groq supported models](https://console.groq.com/docs/models) (suggest you use Llama 3.1 or 3.2)

File: examples/sample_run_local_tools_groq_stream.py [hunk 29]
```diff
@@ -53,7 +55,8 @@ def hello_tool(city: str):
     print(block)
     stream_storage.add(block)
 
+# Run the tools using the Toolhouse client with the created message
 messages += th.run_tools(stream_storage)
 
-response = client.chat.completions.create(model="gpt-4o", messages=messages, tools=th.get_tools() + local_tools)
+response = client.chat.completions.create(model="gpt-4o", messages=messages, tools=th.get_tools() + local_tools) # Retrieve tools installed from Toolhouse
```
Comment (iamdaniele) [inline examples/sample_run_local_tools_groq_stream.py]: `gpt-4o` is an OpenAI model but the code uses Groq, and Groq does not serve OpenAI models. You should change this to any of the [Groq supported models](https://console.groq.com/docs/models) (suggest you use Llama 3.1 or 3.2)

File: examples/sample_run_local_tools_openai.py [hunk 30]
```diff
@@ -3,11 +3,19 @@
...
-from toolhouse import Toolhouse
+from toolhouse import Toolhouse # Import the Toolhouse SDK
 
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+OAI_KEY = os.getenv("OPENAI_API_KEY")
+
+client = OpenAI(access_token=OAI_KEY)
```
Comment (iamdaniele) [inline examples/sample_run_local_tools_openai.py]: `client = OpenAI(api_key=OAI_KEY)`

File: examples/sample_run_local_tools_openai_stream.py [hunk 31]
```diff
@@ -1,15 +1,22 @@
...
+from toolhouse import Toolhouse # Import the Toolhouse SDK
 from toolhouse.models.Stream import ToolhouseStreamStorage
 
 #  Make sure to set up the .env file according to the .env.example file.
 load_dotenv()
 
+TH_API_KEY = os.getenv("TOOLHOUSE_API_KEY")
+OAI_KEY = os.getenv("OPENAI_API_KEY")
+
+client = OpenAI(access_token=OAI_KEY)
```
Comment (iamdaniele) [inline examples/sample_run_local_tools_openai_stream.py]: `client = OpenAI(api_key=OAI_KEY)`

**Review by iamdaniele — CHANGES_REQUESTED** [review]
There are a bunch of files where `access_token` should be `api_key` instead.

Also note that SDKs are smart and can autodetect their API key if they're already set up as environment variables. This removes the need to call `os.getenv` and explicitly set an API Key if they already have the following env variables set up in their system:

- `TOOLHOUSE_API_KEY` (Toolhouse)
- `OPENAI_API_KEY` (OpenAI)
- `ANTHROPIC_API_KEY` (Anthropic)
- `GROQCLOUD_API_KEY` (Groq)

**Comment by aaishikasb** [comment]
Addressed the requested changes in the aforementioned commits!

**Comment by VinciGit00** [comment]
> GitHub

can I have your discord username?

**Review by orliesaurus — APPROVED** [review]
Approved

**Pull request merged**